src/models/dashboard.js

```diff
@@ -189,26 +189,30 @@ export default {
 
       response.forEach(timeseriesResponse => {
         const timeseriesCollection = [];
-        const firstResponse = timeseriesResponse.hits.hits[0]._source;
-        const runId = firstResponse.run.id;
-        const primaryMetric = firstResponse.sample.measurement_title;
-        const iterationName = firstResponse.iteration.name;
-        const sampleName = firstResponse.sample.name;
-        timeseriesResponse.hits.hits.forEach(timeseries => {
-          timeseriesCollection.push({
-            x: timeseries._source['@timestamp_original'],
-            [`y-${runId}_${iterationName}_${sampleName}`]: timeseries._source.result.value,
+        // gets firstResponse only when there is a
+        // non empty array of hits.
+        if (timeseriesResponse.hits.hits.length > 0) {
```

Comment:
Do we want to consider being even more "paranoid" here, not assuming that we necessarily even have a "standard" Elasticsearch success payload here? This will fail, for example, if the payload doesn't have "hits" because what we got was an Elasticsearch error message document. That seems to be what you said you were seeing, but this code doesn't address that possibility.